report/formatted.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package report
+
+import (
+	"errors"
+
+	"github.com/uber-go/gwr/source"
+)
+
+var errReporterClosed = errors.New("reporter closed")
+var errRawSource = errors.New("raw sources unsupported, only item data sources")
+
+// FormattedReporter reports observed items from a data source to a formatting
+// function.  Only works with sources that support the "text" format.
+//
+// For example to send a source to stdandard output:
+//     rep := NewPrintfReporter(someSource, fmt.Printf)
+//     rep.Start() // TODO: error check
+//     defer rep.Stop()
+type FormattedReporter interface {
+	source.ItemWatcher
+	Source() source.DataSource
+	Start() error
+	Stop()
+}
+
+// LogfReporter is a FormattedReporter that targets a log formatting function.
+type LogfReporter struct {
```

Comment:
Do you really need to export the logf reporter and printf reporter types? All their fields are private.

Consider making these private and changing the constructors to return a `FormattedReporter` rather than a `*{Logf,Printf}Reporter`. You can always expose the underlying types later if needed.